### DISCUSSION TITLE
Installing  npm-upgrade

### DISCUSSION THREAD

**Discussion by ghost** [npm/feedback#618] [CLI]
When I run npm i -g npm-upgrade, I get the following error:

641 silly get     'x-amz-replication-status': 'COMPLETED',
641 silly get     server: 'cloudflare',
641 silly get     'content-encoding': 'gzip' } ]
642 verbose get saving yargs to /home/shuller/.npm/registry.npmjs.org/yargs/.cache.json
643 info retry fetch attempt 1 at 12:29:08 PM
644 info attempt registry request try #1 at 12:29:08 PM
645 http fetch GET https://registry.npmjs.org/npm-check-updates/-/npm-check-updates-11.1.9.tgz
646 verbose stack Error: Missing required argument #1
646 verbose stack     at andLogAndFinish (/usr/share/npm/lib/fetch-package-metadata.js:31:3)
646 verbose stack     at fetchPackageMetadata (/usr/share/npm/lib/fetch-package-metadata.js:51:22)
646 verbose stack     at resolveWithNewModule (/usr/share/npm/lib/install/deps.js:456:12)
646 verbose stack     at /usr/share/npm/lib/install/deps.js:457:7
646 verbose stack     at /usr/share/npm/node_modules/iferr/index.js:13:50
646 verbose stack     at /usr/share/npm/lib/fetch-package-metadata.js:37:12
646 verbose stack     at addRequestedAndFinish (/usr/share/npm/lib/fetch-package-metadata.js:82:5)
646 verbose stack     at returnAndAddMetadata (/usr/share/npm/lib/fetch-package-metadata.js:117:7)
646 verbose stack     at pickVersionFromRegistryDocument (/usr/share/npm/lib/fetch-package-metadata.js:134:20)
646 verbose stack     at /usr/share/npm/node_modules/iferr/index.js:13:50
647 verbose cwd /home/shuller
648 error Linux 4.15.0-163-generic
649 error argv "/usr/bin/node" "/usr/bin/npm" "i" "-g" "npm-upgrade"
650 error node v8.10.0
651 error npm  v3.5.2
652 error code EMISSINGARG
653 error typeerror Error: Missing required argument #1
653 error typeerror     at andLogAndFinish (/usr/share/npm/lib/fetch-package-metadata.js:31:3)
653 error typeerror     at fetchPackageMetadata (/usr/share/npm/lib/fetch-package-metadata.js:51:22)
653 error typeerror     at resolveWithNewModule (/usr/share/npm/lib/install/deps.js:456:12)
653 error typeerror     at /usr/share/npm/lib/install/deps.js:457:7
653 error typeerror     at /usr/share/npm/node_modules/iferr/index.js:13:50
653 error typeerror     at /usr/share/npm/lib/fetch-package-metadata.js:37:12
653 error typeerror     at addRequestedAndFinish (/usr/share/npm/lib/fetch-package-metadata.js:82:5)
653 error typeerror     at returnAndAddMetadata (/usr/share/npm/lib/fetch-package-metadata.js:117:7)
653 error typeerror     at pickVersionFromRegistryDocument (/usr/share/npm/lib/fetch-package-metadata.js:134:20)
653 error typeerror     at /usr/share/npm/node_modules/iferr/index.js:13:50
654 error typeerror This is an error with npm itself. Please report this error at:
654 error typeerror     <http://github.com/npm/npm/issues>
655 verbose exit [ 1, true ]

**Comment by ljharb** [comment]
You’re using npm 3 and node 8, both of which are EOL for a long time. Try node 12+, and npm 6 or 8.
